sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_link_async.py

```diff
@@ -182,11 +182,11 @@ async def _incoming_attach(self, frame):
         if self.network_trace:
             _LOGGER.info("<- %r", AttachFrame(*frame), extra=self.network_trace_params)
         if self._is_closed:
-            raise ValueError("Invalid link")
+            raise AMQPLinkError(condition=ErrorCondition.LinkDetachForced, description="Invalid link", info=None)
```

Comment:
I think this should probably be a ClientError.
LinkDetachForced only makes sense if it's coming from or going over the wire.

Though what sets `self._is_closed`? And where else is it checked?

Reply:
`self._is_closed` is set by ` _outgoing_detach(self, close=False, error=None)` when close is set to True

It is then checked here:
*`_check_if_closed`
*`_on_session_state_change`
*`_incoming_attach`
*`_incoming_detach`
*`attach`

I changed it to ClientError because the Link was not forcibly closed but was already closed.